### PR TITLE
Add string interpolation highlighting

### PR DESCRIPTION
**Pull request by PanAeon**
My take on #11
Dark screenshot:
Dark
Light screenshot:
White

One thing I don't like is that `"${variable}"` and `"$variable"` are highlighted differently. ~~I can change it to no highlighting so it would be more consistent.~~
Decided that more consistent syntax is better:

Screen Shot 2019-05-09 at 16 49 19pm

What do you think?

### CODE REVIEW

**Comment by torkelrogstad** [comment]
Tested locally and verified that multi line strings also work.

**Comment by olafurpg** [comment]
Thank you for this contribution! I am super excited to enjoy this fix since the current syntax highlighting makes it really difficult to work with interpolator strings.

I just tried this locally and the quotes `"` for interpolator strings are colored differently from normal strings.

Screenshot 2019-05-13 at 15 26 53

I expected the color for quotes `"` to be colored the same like in the screenshots from @PanAeon in the PR description. I'm using the default dark theme, but I can reproduce the issue with other color themes.

File: src/typescript/Scala.tmLanguage.ts [hunk 1]
```diff
@@ -1,6 +1,15 @@
 "use strict";
 import { TmLanguage } from "./TMLanguageModel";
 
+const letter = "[_a-zA-Z\\$\\p{Lo}\\p{Lt}\\p{Nl}\\p{Ll}\\p{Lu}]"
+const digit  = "[0-9]"
+const letterOrDigit = `${letter}|${digit}`
+const opchar = `[!#%&*+\\-\\/:<>=?@\\\\^|~[\\p{Sm}\\p{So}]]`
+const op     = `${opchar}+`
+const idrest = `${letter}(?:${letterOrDigit})*(?:_${op})?`
```
Comment (olafurpg) [inline src/typescript/Scala.tmLanguage.ts]: This one seems unused

**Comment by PanAeon** [comment]
Good catch. I reproduced the quotes bug. It looks like I mark the text inside string as `string.quoted.{single|tripple}.scala`, but don't do this for the quotes.

File: src/typescript/Scala.tmLanguage.ts (hunk 1 above)
Comment (PanAeon) [reply src/typescript/Scala.tmLanguage.ts]: removed unused declarations

**Comment by olafurpg** [comment]
I believe we should also consider updating the GitHub Linguist highlighting to use this improvements once merged. Code fences on github.com render string interpolators incorrectly

Screenshot 2019-05-14 at 11 12 29

iiuc, this fix could improve that well

**Comment by olafurpg** [comment]
Released as 0.3.1 🎉  https://github.com/scala/vscode-scala-syntax/releases/tag/v0.3.1

**Pull request merged**